src/sonic-yang-models/yang-models/sonic-copp.yang

```diff
@@ -0,0 +1,243 @@
+						policer.  This value represents the depth of the token
+						bucket.";
+
+					must "((current()/../cir) and (current()/../cir > 0))" {
+						error-message "cbs can't be configured without cir.";
+					}
+					must "(current() >= current()/../cir)" {
+						error-message "cbs must be greater than or equal to cir";
+					}
+				}
+
+				leaf pir {
+					type uint64;
+					description
+						"Peak information rate for the dual-rate token bucket
+						policer.  This value represents the rate at which tokens
+						are added to the secondary bucket.";
+
+					must "((current()/../cir) and (current()/../cir > 0))" {
+						error-message "pir can't be configured without cir.";
+					}
+					must "(current() >= current()/../cir)" {
+						error-message "pir must be greater than or equal to cir";
+					}
+				}
+
+				leaf pbs {
+					type uint64;
+					description
+						"Excess burst size for the dual-rate token bucket policer.
+						This value represents the depth of the secondary bucket.";
+
+					must "((current()/../pir) and (current()/../pir > 0))" {
+						error-message "pbs can't be configured without pir.";
+					}
+					must "(current() >= current()/../pir)" {
+						error-message "pbs must be greater than or equal to pir";
+					}
+					must "((not(current()/../cbs)) or (current() >= current()/../cbs))" {
+						error-message "pbs must be greater than or equal to cbs";
+					}
+				}
+				
+				leaf green_action {
+					type string {
+						pattern "(drop)|(forward)|(copy)|(copy_cancel)|(trap)|(log)|(deny)|(transit)" {
+							error-message "Invalid value passed for green_action";
+						}
+					}
+					description
+						"Green action";
+				}
+				
+				leaf yellow_action {
+					type string {
+						pattern "(drop)|(forward)|(copy)|(copy_cancel)|(trap)|(log)|(deny)|(transit)" {
+							error-message "Invalid value passed for yellow_action";
+						}
+					}
+					description
+						"Yellow action";
+				}
+
+				leaf red_action {
+					type string {
+						pattern "(drop)|(forward)|(copy)|(copy_cancel)|(trap)|(log)|(deny)|(transit)" {
+							error-message "Invalid value passed for red_action";
+						}
+					}
+					description
+						"Red action";
+				}
+				
+				leaf genetlink_name {
+					type string {
+						pattern "(psample)" {
+							error-message "Invalid value passed for genetlink_name";
+						}
+					}
+					description
+						"genetlink name";
+				}
+				
+				leaf genetlink_mcgrp_name {
+					type string {
+						pattern "(packets)" {
+							error-message "Invalid value passed for genetlink_mcgrp_name";
+						}
+					}
+					description
+						"genetlink multicast group name";
+				}
+			}
+			/* end of list COPP_GROUP_LIST */
+		}
+		/* end of container COPP_GROUP */
+
+		container COPP_TRAP {
+			list COPP_TRAP_LIST {
+				key name;
+
+				leaf name {
+					type string;
+					description
+						"Control plane policing trap name";
+				}
+
+				leaf trap_ids {
+					type string;
```

Comment:
CoPP HLD expects the trap_ids name to be specific string. Should we define a pattern here, something like [this](https://github.com/Azure/sonic-swss/blob/master/orchagent/copporch.cpp#L39)?

Reply:
trap_ids is a string with comma separated trap_ids.
```
"arp": {
            "trap_ids": "arp_req,arp_resp,neigh_discovery",
            "trap_group": "queue4_group3"
 },
```
If we have to define a pattern based on [copporch](https://github.com/Azure/sonic-swss/blob/master/orchagent/copporch.cpp#L39), 
trap_ids can be defined as leaf-list in yang i.e array.

```
leaf-list trap_ids {
                    mandatory true;
                    type enumeration {
                        enum arp_req;
                        enum arp_resp;
                        enum arp_suppress;
                        ...
                        ...
                        enum vrrp;
                        enum vrrpv6;
                    }
                    description "list of trap_ids";
            }
```

But, this would also mandate
1. changes to [copp_cfg.j2](https://github.com/Azure/sonic-buildimage/blob/master/files/image_config/copp/copp_cfg.j2#L74) trap_ids fields as below: 

```
"arp": {
            "trap_ids": [
                "arp_req",
                "arp_resp",
                "neigh_discovery"
            ],
            "trap_group": "queue4_group2"
        },
```

2. changes to [coppmgr](https://github.com/Azure/sonic-swss/blob/master/cfgmgr/coppmgr.cpp) to handle array of trap_ids. Otherwise, it results into  `ERR swss#coppmgrd: :- main: Runtime error: type must be string, but is array`
3.

Reply:
ok. im not expecting a change to coppmgr or template.